### PR TITLE
[HW9] [js_hw7 'Classes' task_01,task_02]

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: js_hw_7/task_02.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+};
+
+class Snake extends Animal {
+    #isPoisonous
+    constructor (type, color, weight, height, placeOfOrigin, isPoisonous) {
+        super (type, color, weight, height, placeOfOrigin);
+        this.#isPoisonous = isPoisonous;
+    };
+    checkPoisonous(){
+       if (!(typeof this.#isPoisonous === 'boolean')) throw new Error ('isPoisonous should be boolean value');
```
Comment (anatoly-karpovich) [inline js_hw_7/task_02.js]: This check should be performed in counstructor when setting isPoisonous value

File: js_hw_7/task_02.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+// 7. Добавьте метод addAnimal(animal), добавляющий животное в массив animals.
+//     На вход метод должен принимать объект класса Animal, как и любого из его наследников. 
+//     Если объект не является инстансом класса Animal - выкинуть ошибку
+//     ТАКЖЕ, если объект является инстансом класса Snake - выкинуть ошибку с тексом "There will be no snakes, mister Potter!"
+
+// 8. Добавьте методы removeWorker() и removeAnimal() // Подумайте, как будем удалять, по какому полю будем выбирать:)
+
+class Zoo {
+    constructor (address, title, ticketPrice) {
```
Comment (anatoly-karpovich) [inline js_hw_7/task_02.js]: animals and workers can be probably added in the constructor, so maybe it's ok to handle it's arrays in constructor too)

**Review by anatoly-karpovich — APPROVED** [review]
Amaizing job!

File: js_hw_7/task_02.js (hunk 1 above)
Comment (Maximus2806) [reply js_hw_7/task_02.js]: Done

File: js_hw_7/task_02.js (hunk 2 above)
Comment (Maximus2806) [reply js_hw_7/task_02.js]: Implemented such way:
class Zoo {
    constructor (address, title, ticketPrice, workers = [], animals = []) {
        this.address = address;
        this.title = title;
        this.ticketPrice = ticketPrice;
        this.workers = workers;        
        this.animals = animals;        
    };

**Pull request merged**